task/clamav-scan/0.1/clamav-scan.yaml

```diff
@@ -63,40 +65,101 @@ spec:
-
-        EC_EXPERIMENTAL=1 ec test \
-          --namespace required_checks \
-          --policy /project/clamav/virus-check.rego \
-          -o json \
-          /tekton/home/clamscan-result-log.json || true
-
-        # workaround: due to a bug in ec-cli, we cannot generate json and appstudio output at the same time, running it again
-        EC_EXPERIMENTAL=1 ec test \
-          --namespace required_checks \
-          --policy /project/clamav/virus-check.rego \
-          -o appstudio \
-          /tekton/home/clamscan-result-log.json | tee $(results.TEST_OUTPUT.path) || true
+        for file in /tekton/home/clamscan-result-*.log; do
+          if [ -e "$file" ]; then
+            file_suffix=$(basename "$file" | sed 's/clamscan-result-//;s/.log//')
+            # OPA/EC requires structured data input, add clamAV log into json
+            jq -Rs '{ output: . }' /tekton/home/clamscan-result-$file_suffix.log > /tekton/home/clamscan-result-log-$file_suffix.json
+
+            EC_EXPERIMENTAL=1 ec test \
+              --namespace required_checks \
+              --policy /project/clamav/virus-check.rego \
+              -o json \
+              /tekton/home/clamscan-result-log-$file_suffix.json || true
+
+            # workaround: due to a bug in ec-cli, we cannot generate json and appstudio output at the same time, running it again
+            EC_EXPERIMENTAL=1 ec test \
+              --namespace required_checks \
+              --policy /project/clamav/virus-check.rego \
+              -o appstudio \
+              /tekton/home/clamscan-result-log-$file_suffix.json | tee /tekton/home/clamscan-ec-test-$file_suffix.json || true
+
+            cat /tekton/home/clamscan-ec-test-$file_suffix.json
+          fi
+        done
+
+        jq -s -rce '
+          reduce .[] as $item ({"timestamp":"0","namespace":"","successes":0,"failures":0,"warnings":0,"result":"SUCCESS","note":"All checks passed successfully"};
+            {
+            "timestamp" : (if .timestamp < $item.timestamp then $item.timestamp else .timestamp end),
+            "namespace" : $item.namespace,
+            "successes" : (.successes + $item.successes),
+            "failures" : (.failures + $item.failures),
+            "warnings" : (.warnings + $item.warnings),
+            "result" : (if $item.result != "SUCCESS" then $item.result else .result end),
+            "note" : (if $item.result != "SUCCESS" then $item.note else .note end)
+            })' /tekton/home/clamscan-ec-test-*.json | tee $(results.TEST_OUTPUT.path)
+
+        # Add the IMAGE_DIGEST to IMAGES_PROCESSED result string anyways because even if the image index itself is not scanned,
+        # it gets resolved to amd64 arch which has been scanned
+        digests_processed+="${digests_processed:+, }\"$IMAGE_DIGEST\""
+        echo "${images_processed_template/\[%s]/[$digests_processed]}" | tee $(results.IMAGES_PROCESSED.path)
```

Comment:
There is a good practice that script references tekton variable via environment variable.

Reply:
that's for tekton parameters if I'm not mistaken